lib/headcount_analyst.rb

```diff
@@ -1,109 +1,82 @@
+      district_trend[year.to_i] = (district1[year].to_f / district2[year].to_f).round(3)
     end
-    district_trend
-  end
-
-  def district_1_kinder_participation(district)
-    dr.find_by_name(district).enrollment.kindergarten_participation
-  end
-
-  def district_2_kinder_state_participation(district)
-    dr.find_by_name(district[:against]).enrollment.kindergarten_participation
-  end
-
-  def kindergarten_participation_against_high_school_graduation(district, hash = {against: "COLORADO"})
-
-    kinder_variation = kindergarten_participation_rate_variation(district, hash)
-
-    grad_variation = high_school_graduation_variation(district, hash)
-
-    kinder_grad_variance = (kinder_variation/grad_variation).round(3)
-  end
 
-  def hs_1_grad_average(district)
-    dr.find_by_name(district).calculate_hs_grad_average
+    district_trend
   end
 
-  def hs_2_grad_average(district)
-    dr.find_by_name(district[:against]).calculate_hs_grad_average
+  def fetch_district_kindergarten_participation(district_name)
+    district = @district_repository.find_by_name(district_name)
+    district.enrollment_data.kindergarten_participation
   end
 
-  def high_school_graduation_variation(district_1, district_2)
-    hs_1 = hs_1_grad_average(district_1)
-
-    hs_2 = hs_2_grad_average(district_2)
-
-    variation = (hs_1/hs_2).round(3)
+  def does_kindergarten_participation_correlate_with_high_school_graduation(district1_name, district2_name = "COLORADO")
+    does_it_correlate = calculate_data_variation(district1_name, district2_name, :high_school_graduation_rates)
+    does_it_correlate.between?(0.6, 1.5)
   end
 
-  def kindergarten_participation_correlates_with_high_school_graduation(for_hash, hash = {against: "COLORADO"})
-    if for_hash[:for] == 'STATEWIDE'
-      statewide(hash)
-    elsif for_hash[:across].class == Array
-      subset_of_districts(for_hash, hash)
-    else
-      does_it_correlate = high_school_graduation_variation(for_hash[:for], hash)
-      does_it_correlate.between?(0.6,1.5)
+  def does_kindergarten_participation_correlate_with_high_school_graduation_across_subset_of_districts(across_array, against_district)
+    x = across_array.map do |district|
+      does_kindergarten_participation_correlate_with_high_school_graduation(district, against_district)
     end
-  end
 
-  def subset_of_districts(for_hash, hash)
-    districts_to_test = for_hash[:across]
-    x = districts_to_test.map do |district|
-      variations = high_school_graduation_variation(district, hash)
-      variations.between?(0.6,1.5)
-    end
-    if x.count(true)/x.count > 0.700
+    if (x.count(true) / x.count) > 0.700
       true
     else
       false
     end
   end
 
-  def statewide(hash)
-    all_districts = dr.districts.map do |key, value|
-      district_in_string = value.name
-      high_school_graduation_variation(district_in_string, hash)
+  def does_statewide_kindergarten_participation_correlate_with_district_hs_graduation(district2)
```

Comment:
The names of this function and the one above it, while breaking our 80 character limit, are descriptive and clearer than their predecessors.